Repository: plone.recipe.zeoserver


Branch: refs/heads/master
Date: 2020-08-13T08:20:17+05:30
Author: Fazal S (fazalsulaiman) <fazal.sulaiman@gmail.com>
Commit: https://github.com/plone/plone.recipe.zeoserver/commit/d05c5ee99d683ce89ec1a429ba4477a4555f12bf

Fix for "ValueError: Attempted relative import in non-package" - In windows system

Files changed:
A news/42.bugfix
M src/plone/recipe/zeoserver/runzeo.bat
M src/plone/recipe/zeoserver/zeoservice.py.in

b'diff --git a/news/42.bugfix b/news/42.bugfix\nnew file mode 100644\nindex 0000000..965bb0d\n--- /dev/null\n+++ b/news/42.bugfix\n@@ -0,0 +1 @@\n+fix for "ValueError: Attempted relative import in non-package" - In windows build\ndiff --git a/src/plone/recipe/zeoserver/runzeo.bat b/src/plone/recipe/zeoserver/runzeo.bat\nindex cef0f6d..908d7b2 100644\n--- a/src/plone/recipe/zeoserver/runzeo.bat\n+++ b/src/plone/recipe/zeoserver/runzeo.bat\n@@ -3,5 +3,5 @@\n @set ZODB_HOME=%(zodb3_home)s\n @set CONFIG_FILE=%(INSTANCE_HOME)s/etc/%(PACKAGE)s.conf\n @set PYTHONPATH=%(PYTHONPATH)s\n-@set RUNZEO=%%ZODB_HOME%%/ZEO/runzeo.py\n-"%%PYTHON%%" "%%RUNZEO%%" -C "%%CONFIG_FILE%%" %%1 %%2 %%3 %%4 %%5 %%6 %%7 %%8 %%9\n+@set RUNZEO=ZEO.runzeo\n+"%%PYTHON%%" -m "%%RUNZEO%%" -C "%%CONFIG_FILE%%" %%1 %%2 %%3 %%4 %%5 %%6 %%7 %%8 %%9\ndiff --git a/src/plone/recipe/zeoserver/zeoservice.py.in b/src/plone/recipe/zeoserver/zeoservice.py.in\nindex d2ad0e1..5699443 100644\n--- a/src/plone/recipe/zeoserver/zeoservice.py.in\n+++ b/src/plone/recipe/zeoserver/zeoservice.py.in\n@@ -74,7 +74,7 @@ ZODB_HOME = r\'%(zodb_home)s\'\n PYTHONPATH = r\'%(PYTHONPATH)s\'\n PID_FILENAME = r\'%(PID_FILENAME)s\'\n \n-ZEO_RUN = r\'%%s\\\\ZEO\\\\runzeo.py\' %% ZODB_HOME\n+ZEO_RUN = r\'ZEO.runzeo\'\n CONFIG_FILE= os.path.join(INSTANCE_HOME, \'etc\', \'zeo.conf\')\n \n import win32service\n@@ -105,7 +105,7 @@ class InstanceService(Service):\n         _exe_name_ = PYTHONSERVICE_EXE\n \n     process_runner = PYTHON\n-    process_args = \'"%%s" -C "%%s"\' %% (ZEO_RUN, CONFIG_FILE)\n+    process_args = \'-m "%%s" -C "%%s"\' %% (ZEO_RUN, CONFIG_FILE)\n     def __init__(self, args):\n         self.setReg(\'command\', self.process_runner + " " + self.process_args)\n         self.setReg(\'pid_filename\', PID_FILENAME)\n'

Repository: plone.recipe.zeoserver


Branch: refs/heads/master
Date: 2020-08-13T14:13:03+05:30
Author: Fazal Sulaiman (fazalsulaiman) <fazal.sulaiman@gmail.com>
Commit: https://github.com/plone/plone.recipe.zeoserver/commit/abb097f42b59c4e85f042e2aba0c13ec3b711726

Merge pull request #43 from fazalsulaiman/master

Fix for "ValueError: Attempted relative import in non-package" - In windows system

Files changed:
A news/42.bugfix
M src/plone/recipe/zeoserver/runzeo.bat
M src/plone/recipe/zeoserver/zeoservice.py.in

b'diff --git a/news/42.bugfix b/news/42.bugfix\nnew file mode 100644\nindex 0000000..965bb0d\n--- /dev/null\n+++ b/news/42.bugfix\n@@ -0,0 +1 @@\n+fix for "ValueError: Attempted relative import in non-package" - In windows build\ndiff --git a/src/plone/recipe/zeoserver/runzeo.bat b/src/plone/recipe/zeoserver/runzeo.bat\nindex cef0f6d..908d7b2 100644\n--- a/src/plone/recipe/zeoserver/runzeo.bat\n+++ b/src/plone/recipe/zeoserver/runzeo.bat\n@@ -3,5 +3,5 @@\n @set ZODB_HOME=%(zodb3_home)s\n @set CONFIG_FILE=%(INSTANCE_HOME)s/etc/%(PACKAGE)s.conf\n @set PYTHONPATH=%(PYTHONPATH)s\n-@set RUNZEO=%%ZODB_HOME%%/ZEO/runzeo.py\n-"%%PYTHON%%" "%%RUNZEO%%" -C "%%CONFIG_FILE%%" %%1 %%2 %%3 %%4 %%5 %%6 %%7 %%8 %%9\n+@set RUNZEO=ZEO.runzeo\n+"%%PYTHON%%" -m "%%RUNZEO%%" -C "%%CONFIG_FILE%%" %%1 %%2 %%3 %%4 %%5 %%6 %%7 %%8 %%9\ndiff --git a/src/plone/recipe/zeoserver/zeoservice.py.in b/src/plone/recipe/zeoserver/zeoservice.py.in\nindex d2ad0e1..5699443 100644\n--- a/src/plone/recipe/zeoserver/zeoservice.py.in\n+++ b/src/plone/recipe/zeoserver/zeoservice.py.in\n@@ -74,7 +74,7 @@ ZODB_HOME = r\'%(zodb_home)s\'\n PYTHONPATH = r\'%(PYTHONPATH)s\'\n PID_FILENAME = r\'%(PID_FILENAME)s\'\n \n-ZEO_RUN = r\'%%s\\\\ZEO\\\\runzeo.py\' %% ZODB_HOME\n+ZEO_RUN = r\'ZEO.runzeo\'\n CONFIG_FILE= os.path.join(INSTANCE_HOME, \'etc\', \'zeo.conf\')\n \n import win32service\n@@ -105,7 +105,7 @@ class InstanceService(Service):\n         _exe_name_ = PYTHONSERVICE_EXE\n \n     process_runner = PYTHON\n-    process_args = \'"%%s" -C "%%s"\' %% (ZEO_RUN, CONFIG_FILE)\n+    process_args = \'-m "%%s" -C "%%s"\' %% (ZEO_RUN, CONFIG_FILE)\n     def __init__(self, args):\n         self.setReg(\'command\', self.process_runner + " " + self.process_args)\n         self.setReg(\'pid_filename\', PID_FILENAME)\n'

